### PR TITLE
feat(data): Implement DELETE /event/device/name/{name} V2 API

### PR DESCRIPTION
**Pull request by judehung**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## Issue Number:
#2837

## What is the new behavior?
Implement DELETE /event/device/name/{name} for V2 API. The current V2 API doc still lists DELETE /event/device/{deviceId}; however, as https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/models/event.go#L14 will be updated to DeviceName instead of DeviceId, the V2 API spec shall be updated to reflect this change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by judehung** [comment]
recheck

**Comment by judehung** [comment]
@lenny-intel and @cloudxxx8 I've implemented DELETE /event/device/name/{name} for V2 API.  As this PR is on top of codes from https://github.com/edgexfoundry/edgex-go/pull/2840 that you had previously approved, you could simply check the [second commit](https://github.com/edgexfoundry/edgex-go/pull/2874/commits/fb5fae2721e61636878913a70a8c9d88e528f913) to expedite the review.

File: internal/core/data/v2/application/event.go [hunk 1]
```diff
@@ -158,6 +157,32 @@ func EventCountByDevice(deviceName string, dic *di.Container) (uint32, errors.Ed
...
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	return nil
+}
+
+// The DeleteEventsByDeviceName function will be invoked by controller functions
+// and then invokes DeleteEventsByDeviceName function in the infrastructure layer to remove
+// all events/readings that are associated with the given deviceName
+func DeleteEventsByDeviceName(deviceName string, dic *di.Container) errors.EdgeX {
+	dbClient := v2DataContainer.DBClientFrom(dic.Get)
```
Comment (lenny-intel) [inline internal/core/data/v2/application/event.go:177]: Should validate `deviceName` is not empty?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/core/data/v2/application/event.go (hunk 1 above)
Comment (judehung) [reply internal/core/data/v2/application/event.go:181]: @lenny-intel Thanks for the reminder.  I add validation to check if device name is empty or not.  I also added this validation as part of unit tests.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Comment by judehung** [comment]
rebased and squashed all commits into one.

File: openapi/v2/core-data.yaml [hunk 2]
```diff
@@ -1042,11 +993,58 @@ paths:
...
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BaseResponse'
+              examples:
+                200Example:
+                  $ref: '#/components/examples/200Example'
+        '404':
```
Comment (cloudxxx8) [inline openapi/v2/core-data.yaml:1024]: There is no 404 in this API

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: openapi/v2/core-data.yaml (hunk 2 above)
Comment (judehung) [reply openapi/v2/core-data.yaml]: Thanks for the catch.  Removed.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Review by lenny-intel — APPROVED** [review]

**Pull request merged**